jvm-libs/linea/clients/eth-logs-searcher/src/main/kotlin/linea/ethapi/EthLogsSearcherImpl.kt

```diff
@@ -123,8 +124,12 @@ class EthLogsSearcherImpl(
 
         enoughLogsCollected || collectionTimeoutElapsed || noMoreChunksToCollect
       },
+      timeout = searchTimeout,
```

Comment:
I think we should add a timeout here, otherwise, it would keep retrying indefinitely on errors